llrpConnector/src/main/java/io/vantiq/extsrc/llrpConnector/LLRPConnector.java

```diff
@@ -0,0 +1,1546 @@
+
+    // Valid logLevels
+    private static ArrayList<String> LogLevels = new ArrayList<String>(Arrays.asList("error", "warn", "info", "debug"));
+    final static String INFO = "info";
+    final static String WARN = "warn";
+    final static String ERROR = "error";
+    final static String DEBUG = "debug";
+    final static String TAG_DATA_EVENT_NAME = "tagData";
+    final static String READER_DATA_EVENT_NAME = "readerData";
+    final static String READER_STATUS_EVENT_NAME = "readerStatus";
+    final static String M_SUCCESS = "M_Success";  // String value for successful status
+    final static Boolean READER_ONLINE = false;   // Used to set 'readerOffLine' property
+    final static Boolean READER_OFFLINE = true;   // Used to set 'readerOffLine' property
+
+    // Used to reconnect if necessary
+    private String hostname = null;		    // Hostname/IP address of the reader
+    private String sourceName = null;       // Vantiq Source name connected to
+    private int readerPort = -1;			// Reader communication port
+    private int tagReadInterval = 500;		// Frequency for ending AISpec
+    private int logLevel = -1;              // Log level to send to Vantiq
+    private static int ROSPEC_ID = 1;		// ID used for collecting tags
+
+    private Socket connection;				// Socket connection to reader
+    private DataOutputStream out;			// Stream when writing to reader
+    private ReadThread rt = null;			// Reads LLRP msg and add to queue
+    private Thread msgThread = null;		// Process regular msgs from reader
+    private Thread tagThread = null;		// Process tag data msgs from reader
+    private Thread readerAliveThread = null;	// Identify when reader is offline
+    private LLRPConnectorCore llrpCore;
+    private Boolean readerOffline = true;	// Indicates the Reader is Offline
+    private Boolean vantiqSourceOffline = true;	// Indicates the connection to the Vantiq Source is Offline
+    Instant lastKeepAlive = Instant.now();	// Last point we rcvd from reader
+
+    /*
+     * RECONNECT_INTERVAL is the duration to wait to check if Reader is online.
+     * It may take 15-30 seconds to flush through stored messages on the reader
+     * since we do not turn off the reading of tags (stored on reader and
+     * processed on reconnect).
+     * The KEEPALIVE_INTERVAL is the periodic interval the reader will use
+     * to send a KEEPALIVE message.  If we don't receive a KEEPALIVE message
+     * within the RECONNECT_INTERVAL, we drop the connection and try to
+     * reconnect indefinitely.
+     */
+    final static int KEEPALIVE_INTERVAL = 5000;
+    final static int RECONNECT_INTERVAL = 10000;
+
+    /**
+     * Message queues for the tags and other messages
+     */
+    private LinkedBlockingQueue<LLRPMessage> queue
+            = new LinkedBlockingQueue<LLRPMessage>();
+    private LinkedBlockingQueue<LLRPMessage> tagQueue
+            = new LinkedBlockingQueue<LLRPMessage>();
+
```

Comment:
are there or should there be size limits here?  IF the reader malfunction and sends a billion messages, what happens?

Reply:
Good question, what do you recommend?